std/range/package.d

```diff
@@ -5297,285 +5297,279 @@ auto sequence(alias fun, State...)(State args)
     }
     ---
 */
+auto iota(E)(E end)
+{
+    static if (is(typeof(end += 1)))
```

Comment:
This is not good. We should not keep unnecessarily 1 as state and lose the benefit of simpler and faster `++` as opposed to `+= variable`.